File: 链表/202.快乐数/202.java
class Solution {
    public boolean isHappy(int n) {
        int slow=n;
        int fast=getNext(n);
        while(fast!=1 && slow!=fast){
            slow=getNext(slow);
            fast=getNext(getNext(fast));
        }
        return fast==1;
    }

    public int getNext(int n){
        int sum=0;
        while(n>0){
            int d=n%10;
            n=n/10;
            sum+=d*d;
        }
        return sum;
    }
}

File: 链表/202.快乐数/202.js
/**
 * @param {number} n
 * @return {boolean}
 */

function getNext(n) {
    let sum=0;
    while(n>0){
        let d=n%10;
        //n-必须要通过Math.floor取整
        n=Math.floor(n/10);//Math.floor返回小于或等于一个给定数字的最大整数
        sum+=d*d;
    }
    return sum
}
 var isHappy = function(n) {
    let slow=n;
    let fast=getNext(n);
    while(fast!==1 && fast!==slow){
        slow=getNext(slow);
        fast=getNext(getNext(fast));
    }
    return fast==1
};